transform/slice.go

```diff
@@ -0,0 +1,14 @@
+// Copyright 2022 Canonical Ltd.
+// Licensed under the LGPLv3, see LICENCE file for details.
+
+package transform
+
+// Slice transforms a slice of one type to an equal length slice of another,
+// by applying the input transformation function to each member.
+func Slice[F any, T any](from []F, transform func(F) T) []T {
+	to := make([]T, len(from))
```

Comment:
Possible minor optimization would be to check if `len(from) == 0` and return a nil slice if so -- this would avoid a memory allocation in the case where the input was nil or zero length. Alternatively only return nil if `from == nil`.